### PR TITLE
some CI checks

### LINKED ISSUE #26: CI
would it be okay to add CI for testing this project?

### PR DESCRIPTION
**Pull request by Kreijstal**
Here I just limited myself for a basic, fixes #26
`cargo build`
and 
`cargo test`
poppler was the annoying part but it works.

feel free to add a cargo fmt check with
```
    - name: Check code formatting
      run: cargo fmt -- --check
```
or linting
```
     - name: Run Clippy
       run: cargo clippy -- -D warnings
```

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+             -DENABLE_QT6=OFF \
+             -DENABLE_BOOST=OFF \
+             -DENABLE_SPLASH=OFF \
+             -DENABLE_LIBCURL=OFF
+        make -j$(nproc)
+        sudo make install
+        sudo ldconfig
+    - uses: actions/checkout@v4
+    - name: Build
+      run: cargo build --verbose
```
Comment (itsjunetime) [inline .github/workflows/rust.yml:51]: I think we could probably do without the `--verbose` flag, I doubt that much output will be useful. I also think it would be better to run `clippy` instead of `build`, as you suggested.
```suggestion
    - name: Clippy
      run: cargo clippy
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Run tests
+      run: cargo test --verbose
```
Comment (itsjunetime) [inline .github/workflows/rust.yml:53]:
```suggestion
      run: cargo test
```

**Review by itsjunetime — APPROVED** [review]
Overall looks great, thanks for doing this! I have a few comments but mostly it seems good to go

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Setup sccache
+      if: github.event_name != 'release' && github.event_name != 'workflow_dispatch'
+      uses: mozilla-actions/sccache-action@v0.0.6
```
[the remark this answers is not in the document]
Comment (Kreijstal) [reply .github/workflows/rust.yml:20]: The speedup is certainly insane like it builds it in 2 minutes instead of 10, yes it's for free. Cache is emptied automatically by GitHub

[the remark this answers is not in the document]
Comment (Kreijstal) [reply .github/workflows/rust.yml:20]: and in cmake is not transparent, but that's why I use sccache flags, this makes sense, because it compiles poppler almost instantly, which we don't care much about since it's a dependency

**Comment by Kreijstal** [comment]
I don't recommend having test without build:
Example
```
topkek@ayylmai UCRT64 ~/git/tdf
# cargo test --no-default-features
warning: file `E:\msys64\home\topkek\git\tdf\./benches/for_profiling.rs` found to be present in multiple build targets:
  * `bin` target `for_profiling`
  * `bench` target `for_profiling`
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.40s
     Running unittests src\lib.rs (target\debug\deps\tdf-86c3d42ed960db73.exe)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests benches/for_profiling.rs (target\debug\deps\for_profiling-3af58501b3ce0734.exe)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests src/main.rs (target\debug\deps\tdf-48ad9523dce8824a.exe)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests tdf

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


topkek@ayylmai UCRT64 ~/git/tdf
# cargo build --no-default-features
warning: file `E:\msys64\home\topkek\git\tdf\./benches/for_profiling.rs` found to be present in multiple build targets:
  * `bin` target `for_profiling`
  * `bench` target `for_profiling`
   Compiling tdf v0.1.0 (E:\msys64\home\topkek\git\tdf)
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
```

**Comment by Kreijstal** [comment]
check won't check for linking problems

**Comment by itsjunetime** [comment]
Huh, that's interesting. I kinda assumed running tests would ensure linking works but I guess there could be some edge cases. I'll add it back in and merge this.

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+             -DENABLE_SPLASH=OFF \
+             -DENABLE_LIBCURL=OFF
+        make -j$(nproc)
+        sudo make install
+        sudo ldconfig
+    - uses: actions/checkout@v4
+    - name: Clippy
+      run: cargo clippy
+    - name: Run tests
+      run: cargo test
```
Comment (itsjunetime) [inline .github/workflows/rust.yml:53]:
```suggestion
      run: cargo test
    - name: Build
      run: cargo build
```

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+             -DENABLE_QT6=OFF \
+             -DENABLE_BOOST=OFF \
+             -DENABLE_SPLASH=OFF \
+             -DENABLE_LIBCURL=OFF
+        make -j$(nproc)
+        sudo make install
+        sudo ldconfig
+    - uses: actions/checkout@v4
+    - name: Clippy
+      run: cargo clippy
```
Comment (itsjunetime) [inline .github/workflows/rust.yml:51]:
```suggestion
      run: cargo clippy -- -D warnings
    - name: Check fmt
      run: cargo fmt -- --check
```
